src/main/java/net/neoforged/neoforge/client/model/generators/template/FaceBuilder.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.client.model.generators.template;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import net.minecraft.client.data.models.model.TextureSlot;
+import net.minecraft.client.renderer.block.model.BlockElementFace;
+import net.minecraft.client.renderer.block.model.BlockFaceUV;
+import net.minecraft.core.Direction;
+import net.neoforged.neoforge.client.model.ExtraFaceData;
+import org.apache.commons.lang3.mutable.MutableObject;
+import org.jetbrains.annotations.Nullable;
+
+public final class FaceBuilder {
+    @Nullable
+    private Direction cullface;
+    private int tintindex = -1;
+    @Nullable
```

Comment:
Can it really be classified as `Nullable` if it errors when it is null? 
`build()` calls `.toString()`

Reply:
I mean yeah, internally it is null cause it has no initial value (what texture slot would you default every face to?) but all attempts for users to set the texture are null checked and the final `build` call also null checks the texture, making it a required non null field.

this is also how face builder has always functioned in the legacy modelgen code and no ones had issues with it before as far as im aware.